rpcs3/rpcs3qt/rpcn_settings_dialog.cpp

```diff
@@ -66,8 +113,506 @@ rpcn_settings_dialog::rpcn_settings_dialog(QWidget* parent)
+	QHBoxLayout* hbox_lbl_combo = new QHBoxLayout();
+	QLabel* lbl_server          = new QLabel(tr("Server:"));
+	cbx_servers                 = new QComboBox();
+
+	refresh_combobox();
+
+	hbox_lbl_combo->addWidget(lbl_server);
+	hbox_lbl_combo->addWidget(cbx_servers);
+
+	QHBoxLayout* hbox_buttons   = new QHBoxLayout();
+	QPushButton* btn_add_server = new QPushButton(tr("Add"));
+	QPushButton* btn_del_server = new QPushButton(tr("Del"));
+	hbox_buttons->addStretch();
+	hbox_buttons->addWidget(btn_add_server);
+	hbox_buttons->addWidget(btn_del_server);
+
+	vbox_server->addLayout(hbox_lbl_combo);
+	vbox_server->addLayout(hbox_buttons);
+
+	grp_server->setLayout(vbox_server);
+	vbox_global->addWidget(grp_server);
+
+	QGroupBox* grp_buttons    = new QGroupBox();
+	QVBoxLayout* vbox_buttons = new QVBoxLayout();
+	QPushButton* btn_create   = new QPushButton(tr("Create Account"));
+	QPushButton* btn_edit     = new QPushButton(tr("Edit Account"));
+	QPushButton* btn_test     = new QPushButton(tr("Test Account"));
+
+	vbox_buttons->addSpacing(10);
+	vbox_buttons->addWidget(btn_create);
+	vbox_buttons->addSpacing(10);
+	vbox_buttons->addWidget(btn_edit);
+	vbox_buttons->addSpacing(10);
+	vbox_buttons->addWidget(btn_test);
+	vbox_buttons->addSpacing(10);
+	grp_buttons->setLayout(vbox_buttons);
+
+	vbox_global->addWidget(grp_buttons);
+
+	setLayout(vbox_global);
+
+	connect(cbx_servers, QOverload<int>::of(&QComboBox::currentIndexChanged), this, [this]([[maybe_unused]] int index)
+		{
+			QVariant host = cbx_servers->itemData(index);
+
+			if (!host.isValid())
+				return;
+
+			g_cfg_rpcn.set_host(host.toString().toStdString());
+			g_cfg_rpcn.save();
+		});
+
+	connect(btn_add_server, &QAbstractButton::clicked, this, [this]()
+		{
+			rpcn_add_server_dialog dlg(this);
+			dlg.exec();
+			const auto new_server = dlg.get_new_server();
+			if (new_server)
+			{
+				if (!g_cfg_rpcn.add_host(new_server->first, new_server->second))
+				{
+					QMessageBox::critical(this, tr("Existing Server"), tr("You already have a server with this description & hostname in the list."), QMessageBox::Ok);
+					return;
+				}
+
+				g_cfg_rpcn.save();
+				refresh_combobox();
+			}
+		});
+
+	connect(btn_del_server, &QAbstractButton::clicked, this, [this]()
+		{
+			int index = cbx_servers->currentIndex();
+
+			if (index == -1)
+				return;
+
+			const auto desc = cbx_servers->itemText(index).toStdString();
+			const auto host = cbx_servers->itemData(index).toString().toStdString();
+
+			ensure(g_cfg_rpcn.del_host(desc, host));
+			g_cfg_rpcn.save();
+			refresh_combobox();
+		});
+
+	connect(btn_create, &QAbstractButton::clicked, this, [this]()
+		{
+			rpcn_ask_username_dialog dlg_username(this, tr("Please enter your username.\n\n"
+														   "Note that these restrictions apply:\n"
+														   "-Username must be between 3 and 16 characters\n"
+														   "-Username can only contain a-z A-Z 0-9 '-' '_'\n"
+														   "-Username is case sensitive\n"));
```

Comment:
wtf is this formatting XD.
we use spaces for that.
Also, I think that spaces after the - would probably look nicer

Reply:
Please don't make me fight the autoformatter!

Reply:
Because if I do it selectively I got a Megamouse telling me I forgot to format this or that appropriately, so I just click format file :D.